core/arch/arm/mm/tee_pager.c

```diff
@@ -919,18 +930,18 @@ bool tee_pager_set_uta_area_attr(struct user_ta_ctx *utc, vaddr_t base,
 			if (pmem->area != area)
 				continue;
 			area_get_entry(pmem->area, pmem->pgidx, &pa, &a);
-			if (a & TEE_MATTR_VALID_BLOCK)
-				assert(pa == get_pmem_pa(pmem));
-			else
-				pa = get_pmem_pa(pmem);
+			assert(pa == get_pmem_pa(pmem));
```

Comment:
@etienne-lms it seems you're trying to comment on a specific commit here given that the diff is marked as "Outdated". Which commit is quite hard for me to tell even by searching through all diffs for something that matches.

Reply:
For big PRs like this one, it is probably best to make one review per commit.

Reply:
This was related to commit "core: pager: add pmem flags" where ([here](https://github.com/OP-TEE/optee_os/pull/2801/files#diff-5b29521d3e6f0086fdd5670fb29c63dcL1038)):
```patch
 			area_get_entry(pmem->area, pmem->pgidx, &pa, &a);
-			if (a & TEE_MATTR_VALID_BLOCK)
-				assert(pa == get_pmem_pa(pmem));
-			else
-				pa = get_pmem_pa(pmem);
+			assert(pa == get_pmem_pa(pmem));
```
Then commit "core: pager: don't store hidden page pa in table" removes the assertion:
```patch
 			area_get_entry(pmem->area, pmem->pgidx, &pa, &a);
-			assert(pa == get_pmem_pa(pmem));
```
In the end, this is fine since the phys address is no more read from `area_get_entry()` but from `get_pmem_pa(pmem)`. It is just the intermediate commits that maybe fail on `assert(pa == get_pmem_pa());` until it is removed by commit "core: pager: don't store hidden page pa in table".

Reply:
This commit is mostly a transitional commit trying to take as small steps as possible in order to make bisect more useful.
I've run complete xtest with this commit in the past so I know it works. Why I thought it was good to leave the assert to finally remove it in "core: pager: don't store hidden page pa in table" I can't remember any longer.